src/firefly/js/util/VOAnalyzer.js

```diff
@@ -0,0 +1,419 @@
+     * @param candidatePairs
+     * @returns {null|{lonCol: *, latCol: *, lonIdx: (number|*|lonIdx), latIdx: (number|*|latIdx), csys: *}|*}
+     */
+    getCenterColumnsOnObsCoreUType(candidatePairs) {
+        this.centerColumnsInfo = null;
+
+        const colPairs = this.getCenterColumnPairsOnUType(candidatePairs);
+
+        if (colPairs && colPairs.length === 1) {
+            this.setCenterColumnsInfo(colPairs[0], posCol[UCDCoord.eq.key].coord);
+        }
+        this.centerColumnCandidatePairs = colPairs;
+
+        return this.centerColumnsInfo;
+    }
+
+    /**
+     * search center column pair by checking ObsCore columns on existing candidate pairs or all table columns
+     * @param candidatePairs
+     * @returns {null|{lonCol: *, latCol: *, lonIdx: *, latIdx: *, csys: *}|*}
+     */
+    getCenterColumnsOnObsCoreName(candidatePairs) {
+        this.centerColumnsInfo = null;
+
+        const leftMostCol = (isEmpty(candidatePairs))
+                            ? null : candidatePairs[0];
+
+        const colPair = this.getCenterColumnPairOnName(candidatePairs);
+
+        if (isArray(colPair) && colPair.length === 2) {
+            return this.setCenterColumnsInfo(colPair, posCol[UCDCoord.eq.key].coord);
+        } else {
+            return leftMostCol?
+                   this.setCenterColumnsInfo(leftMostCol, posCol[UCDCoord.eq.key].coord) :
+                   this.centerColumnsInfo;
+        }
+    }
+
+    /**
+     * search center columns pair by guessing the column name
+     * @returns {null|{lonCol: *, latCol: *, lonIdx: *, latIdx: *, csys: *}|*}
+     */
+    guessCenterColumnsByName() {
+        this.centerColumnsInfo = null;
+
+        const guess = (lon, lat) => {
+            const lonCol = getColumn(this.tableModel, lon);
+            const latCol = getColumn(this.tableModel, lat);
+
+            return (lonCol && latCol) ? this.setCenterColumnsInfo([lonCol, latCol]) : this.centerColumnsInfo;
+        };
+        return (guess('ra','dec') || guess('lon', 'lat'));
+    }
+
+    /**
+     * return center position or catalog coordinate columns and the associate*d coordinate system
+     * by checking table meta, UCD values, Utype, ObsCore column name and guessing.
+     * @returns {{lonCol, latCol, csys}|*}
+     */
+    getCenterColumns() {
+        return  this.getCenterColumnsOnMeta() ||
+                this.getCenterColumnsOnUCD() ||
+                this.getCenterColumnsOnObsCoreUType(this.centerColumnCandidatePairs) ||
+                this.getCenterColumnsOnObsCoreName(this.centerColumnCandidatePairs) ||
+                (isEmpty(this.centerColumnCandidatePairs) && this.guessCenterColumnsByName());
+    }
+
+
+    static newInstance(tableModel) {
+        return new TableRecognizer(tableModel);
+    }
+}
+
+export function findTableCenterColumns(table) {
+   const tblRecog = table && get(table, ['tableData', 'columns']) && TableRecognizer.newInstance(table);
```

Comment:
is there not a TableUtil function to get the columns? If not you should add one an use it here.

Reply:
there was function getCenterColumns in TableInfoUtil.js and in this PR the content of TableInfoUtil.getCenterColumns is replaced by calling the above function, findTableCenterColumns. Portions like catalog watch, coverage watcher, chart util,  lsst footprint or wise plot request all use TableInfoUtil.getCenterColumns to get ra/dec columns.